src/server/room.js

```diff
@@ -44,11 +45,27 @@ class Room {
 
   constructor(id /* : string */) {
     this.id = id;
+    this.playerList = [];
+    this.teamNone = { ...defaultTeam};
     this.teamA = { ...defaultTeam };
     this.teamB = { ...defaultTeam };
     this.phase = 'team-a-coding';
     this.round = 1;
   }
+
+  addPlayerToRoom(userId /*: string */) {
+    if (this.playerList.includes(userId)) {
```

Comment:
fail only if `playerList.includes(userId)` **and** user is active. if user is **not** active, we should flip the active boolean to `true` and return success.